File: sparseArray/sparseArray.ts
const sparseArray2 = (strings:string[],queries:string[]):number[]=>{
   //Como usar Record --> type MyDictionary = Record<Clave, TipoValor>;
    const frecuencyMap: Record<string, number> = {};

   for(const string of strings){
    frecuencyMap[string] = (frecuencyMap[string] || 0)+1
   }

   const result:number[] = [];
   for (const query of queries){
    let frecuency:number;
    frecuency = frecuencyMap[query] || 0;
    result.push(frecuency);
   }

    return result
}

const q2:string[] = ["aba", "xzxb", "ab"];
const s2:string[] = ["aba", "baba", "aba", "xzxb"];

console.log(sparseArray2(s2, q2));

File: sparseArray/SparseArray.java
package sparseArray;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class SparseArray {
    public static List<Integer> sparseArray(String[] strings, String[] queries ){
        Map<String,Integer> frecuencyMap = new HashMap<>();
     for(String string : strings){
        frecuencyMap.put(string,frecuencyMap.getOrDefault(string,0)+1);
        }

    List<Integer> resultList = new ArrayList<>();
    for(String query : queries){
        int frecuency = frecuencyMap.getOrDefault(query,0);
        resultList.add(frecuency); 
    }
        return resultList;
    }

    public static void main (String[] args) {
        String[] q2 = {"aba", "xzxb", "ab"};
        String[] s2 = {"aba", "baba", "aba", "xzxb"};

        sparseArray(s2,q2);
    }
}
